RecoHI/HiEvtPlaneAlgos/src/EvtPlaneProducer.cc

```diff
@@ -305,214 +491,204 @@ void EvtPlaneProducer::produce(edm::Event &iEvent, const edm::EventSetup &iSetup
+      track.eta = trk.eta();
+      track.phi = trk.phi();
+      track.pt = trk.pt();
+      track.ptError = trk.ptError();
+      track.numberOfValidHits = trk.numberOfValidHits();
+      track.algos = trk.algo();
+      track.dz = std::abs(trk.dz(bestvtx));
+      track.dxy = std::abs(trk.dxy(bestvtx));
+      track.dzError = std::hypot(trk.dzError(), bestvzError);
+      track.dxyError = trk.dxyError(bestvtx, vtx_cov);
+      track.dzSig = track.dz / track.dzError;
+      track.dxySig = track.dxy / track.dxyError;
+      const reco::HitPattern &hit_pattern = trk.hitPattern();
+      track.normalizedChi2 = (*chi2Map)[cands->ptrAt(i)];
+      track.chi2layer = (*chi2Map)[cands->ptrAt(i)] / hit_pattern.trackerLayersWithMeasurement();
+      if (cuts->isGoodTrack(track)) {
+        FillTracker(track, bestvz, bin);
       }
     }
-  }
 
-  //Castor part
-
-  iEvent.getByToken(castorToken, castorCollection_);
-
-  if (castorCollection_.isValid()) {
-    for (std::vector<reco::CastorTower>::const_iterator j = castorCollection_->begin(); j != castorCollection_->end();
-         j++) {
-      tower_eta = j->eta();
-      tower_phi = j->phi();
-      tower_energyet = j->et();
-      double minet = minet_;
-      double maxet = maxet_;
-      for (int i = 0; i < NumEPNames; i++) {
-        if (EPDet[i] == Castor) {
-          if (minet_ < 0)
-            minet = minTransverse[i];
-          if (maxet_ < 0)
-            maxet = maxTransverse[i];
-          if (tower_energyet < minet)
-            continue;
-          if (tower_energyet > maxet)
-            continue;
-          double w = tower_energyet;
-          if (EPOrder[i] == 1) {
-            if (MomConsWeight[i][0] == 'y' && loadDB_) {
-              w = flat[i]->getW(tower_energyet, vzr_sell, bin);
-            }
-            if (tower_eta < 0)
-              w = -w;
+    iEvent.getByToken(lostToken_, cands);
+    iEvent.getByToken(chi2MapLostToken_, chi2Map);
+    for (unsigned int i = 0, n = cands->size(); i < n; ++i) {
+      track = {};
+      track.centbin = cbin;
+      if (cuts->isGoodHF(track)) {
+        fillHF(track, bestvz, bin);
+      }
+      const pat::PackedCandidate &pf = (*cands)[i];
+      if (!pf.hasTrackDetails())
+        continue;
+      const reco::Track &trk = pf.pseudoTrack();
+      track.highPurity = pf.trackHighPurity();
+      track.charge = trk.charge();
+      if (!track.highPurity || track.charge == 0)
+        continue;
+      track.collection = 2;
+      track.eta = pf.eta();
+      track.phi = pf.phi();
+      track.et = pf.et();
+      track.pdgid = pf.pdgId();
+      track.pt = trk.pt();
+      track.ptError = trk.ptError();
+      track.numberOfValidHits = trk.numberOfValidHits();
+      track.algos = trk.algo();
+      track.dz = trk.dz(bestvtx);
+      track.dxy = -1. * trk.dxy(bestvtx);
+      track.dzError = sqrt(pow(trk.dzError(), 2) + pow(bestvzError, 2));
+      track.dxyError = sqrt(pow(trk.dxyError(), 2) + bestvxError * bestvyError);
```

Comment:
see above `       track.dxyError = trk.dxyError(bestvtx, vtx_cov);` 
also, refactor to reuse track filling as mentioned above